src/librustc_mir/interpret/validity.rs

```diff
@@ -505,19 +503,20 @@ impl<'rt, 'mir, 'tcx, M: Machine<'mir, 'tcx>> ValueVisitor<'mir, 'tcx, M>
                     // Only NULL is the niche.  So make sure the ptr is NOT NULL.
                     if self.ecx.memory.ptr_may_be_null(ptr) {
                         // These conditions are just here to improve the diagnostics so we can
-                        // differentiate between null pointers and dangling pointers
+                        // differentiate between null pointers and dangling pointers.
                         if self.ref_tracking_for_consts.is_some() &&
-                            self.ecx.memory.get(ptr.alloc_id).is_err() &&
-                            self.ecx.memory.get_fn(ptr).is_err() {
+                            self.ecx.memory.get_size_and_align(ptr.alloc_id, AllocCheck::Live)
+                                .is_err()
+                        {
                             return validation_failure!(
-                                "encountered dangling pointer", self.path
+                                "a dangling pointer", self.path
```

Comment:
Note to self: we don't seem to have a test covering this particular failure. I should try to write one.

Reply:
Turns out this is dead code, I will remove this branch.

@oli-obk if there is a dangling reference, interning will already error and we will never get here.